### DISCUSSION TITLE
Random disconnects

### DISCUSSION THREAD

**Discussion by felixerdy** [capacitor-community/bluetooth-le#681] [Q&A]
Our app is collects data from the BLE periphial as well as geolocation, both in foreground but also in background with the screen turned off. 

After connecting with a BLE periphial, sometimes the connection stops randomly. We tried using different phones, OS, and hardware. 

Sometimes, the connection is pretty stable and we stay connected (tested up to half an hour), but sometimes the connection stops after a few seconds. That's independent of the smartphone, OS, or BLE device we are using. 

Any ideas why these random disconnects happen and how to avoid these to get a stable connection?

**Comment by peitschie** [comment]
When you say "stops"... what kind of error are you actually getting? Is it a write timeout? Does the peripheral disconnect?

**Reply by felixerdy** [reply]
There is no error unfortunately. The [`onDisconnect`](https://github.com/capacitor-community/bluetooth-le?tab=readme-ov-file#connect) callback is called, but no other information. We used different microcontrollers with BLE modules and they all disconnected from our devices after a while. We assume it's because of our app.
Right now, we are experimenting with some kind of automatic reconnection logic that will hopefully establish a new connection if the user did not initiated a disconnect
